manubot/cite/handlers.py

```diff
@@ -98,9 +98,9 @@ class Handler:
     """
 
     prefix_lower: str
-    prefixes = []
+    prefixes: List[str] = dataclasses.field(default_factory=list)
```

Comment:
Cherry picked e8f9e8660f661b6b66b42f5ed8cd12a03fa2773d for https://github.com/manubot/manubot/pull/311 and noticed:

Setting the type annotation to `List[str]` causes the dataclass to consider this a field and causes subclasses to silently fail to set prefixes. For example, the following stops working:

https://github.com/manubot/manubot/blob/b0b0a1873a79dc5ab216fa731ba05817621e1938/manubot/cite/isbn.py#L12-L14

Reply:
Ahh. Then it needs to be `ClassVar[List[str]]` but still I would advise against making the default as a list, since it is mutable and shared among all subclasses

Reply:
Ah good to know about ClassVar. That is indeed what I'm looking for.

So the options are:

```py
# option 1
prefixes: ClassVar[List[str]] = []

# option 2
prefixes: ClassVar[List[str] | None] = None

# option 3
prefixes: ClassVar[List[str]] = dataclasses.field(default_factory=list)
```

Are any of these right? Option 1 has the shared mutability issue. Option 2 requires special casing to treat None as an empty list. Option 3 might be totally wrong because `dataclasses.field` is for instance variables not class variables, right?

Reply:
Option 2 is the way to go

Reply:
Just an extra note - you're right that option 3 is totally wrong. A fourth option would be just to type annotate `prefixes: ClassVar[List[str]]`, since you assume all subclasses _must_ provide this list. Unfortunately there isn't something like "abstract class variable" you can annotate on here, so your options for enforincing this are:

1. Duck typing (just assume it, and some code will fail if it's ever wrong)
2. Overkill write your own metaclass
3. Lightweight check in `__init_subclass__` that with hasattr on the class itself